src/Elements/qualsource.cpp

```diff
@@ -64,7 +64,15 @@ double QualSource::getQuality(Node* node)
 {
     // ... no source contribution if no flow out of node
     quality = node->quality;
-    if ( outflow == 0.0 ) return quality;
+    if (abs(outflow) <= ZERO_FLOW)
```

Comment:
use ZERO_FLOW to determine the stagnant flow but not 0.0.

Reply:
There appears to be a major problem with the original code here -- the value of `QualSource::outflow` is never assigned anywhere. To fix this one has to:
1. Add the following line to ltdsolver.h:
```
std::vector<double>    qOut;            // flow in links leaving each node
```
2. Add the following to the top of LTDSolver::solve:
```
    memset(&qOut, 0, nodeCount*sizeof(double));
    for (int i = 0; i < linkCount; i++) 
    {
            Link* link = network->link(i);
            double q = link->flow;
            int j = link->fromNode->index;
            if ( q < 0.0 ) j = link->toNode->index;
            qOut[j] += abs(q);
    }
```
3. Add the following to LTDSolver::release:
```
    // ... modify node quality c to include any source input
    if ( node->qualSource && network->qualModel->type == QualModel::CHEM )
    {
        node->qualSource->outflow = qOut[node->index];   //new line here
        c = node->qualSource->getQuality(node);
        . . . 
```